src/index.php

```diff
@@ -33,7 +33,8 @@
 try {
     // get streak stats for user given in query string
     $user = preg_replace("/[^a-zA-Z0-9\-]/", "", $_REQUEST["user"]);
-    $contributionGraphs = getContributionGraphs($user);
+    $minContribYear = intval($_REQUEST["override_starting_year"]);
```

Comment:
This will evaluate to `0` if `override_starting_year` is not present in the request (which will make it send 2,023 requests to GitHub). To make it `null` when it is not provided, it should be set to null outside the call to `intval` which always returns an `int`.

```suggestion
    $minContribYear = $_REQUEST["override_starting_year"] ? intval($_REQUEST["override_starting_year"]) : null;
```

Another way using null coalescing operator and ternary shorthand:

```suggestion
    $minContribYear = intval($_REQUEST["override_starting_year"] ?? "") ?: null;
```